cpp/include/cuspatial/detail/nvtx/nvtx3.hpp

```diff
@@ -0,0 +1,1934 @@
+/*
+ *  Copyright (c) 2020-2022, NVIDIA CORPORATION.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
```

Comment:
No, I added these at @isVoid 's request so that when the developer guide refers to NVTX and `CUSPATIAL_FUNC_RANGE` it actually exists. Now we just have to start using them.